mtwitter.js

```diff
@@ -0,0 +1,254 @@
+var Twitter = require('twitter');
+var Request = require('request');
+var fs = require('fs');
+var Buffer = require('buffer').Buffer;
+
+var TWITTER_PROFILE_INTERVAL = 60000;
+
+var mtwitter = function(bridge, config) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tuser_cache = {};
+  this.timeline_list = [];
+  this.timeline_queue = [];
+  this.timeline_period = 0;
+  this.timeline_intervalobj = null;
+  this._bridge = bridge;
+
+  this.get_bearer_token((bt) => {
+      if (bt != null) {
+          this.app_auth.bearer_token = bt;
+          console.log("Twitter Application Auth OK!");
+          this.app_twitter = new Twitter(this.app_auth);
+          return;
+      }
+      console.error("Twitter Application Failed Auth. The bridge will be operating in a limited capacity.");
+  });
+}
+
+mtwitter.prototype.get_bearer_token = function(cb){
+  try {
+      fs.accessSync('bearer.tok', fs.R_OK);
+      var tok = fs.readFileSync('bearer.tok', 'utf8');
+      cb(tok);
+      return;
+  } catch (e) {
+      console.log("Bearer token either does not exist or cannot be read, getting a new token.")
+  }
+
+  var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+  key = Buffer.from(this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret, 'ascii').toString('base64');
+  var options = {
+      url: "https://api.twitter.com/oauth2/token",
+      headers: {
+          'Authorization': "Basic " + key
+      },
+      form: "grant_type=client_credentials",
+      contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+  };
+  Request.post(options, function(error, response, body) {
+      if (error) {
+          console.log("Error", error);
+          return false;
+      } else {
+          try {
+              var jsonresponse = JSON.parse(body);
+              if (jsonresponse.token_type == "bearer") {
+                  fs.writeFile("bearer.tok", jsonresponse.access_token, (err) => {
+                      if (err) {
+                          console.error("Couldn't write bearer token to file. Reason:",err);
+                      }
+                  });
+                  cb(jsonresponse.bearer_token);
```

Comment:
Comments please. It's surprising that you're invoking the callback early rather than waiting for `bearer.tok` to be written to disk first. It's okay to do it like this, but you should explain why (e.g. because if you can't write to file then meh, can always get a new one).